modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -583,23 +583,27 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
         Ptr<Impl> impl;
     };
 
-    /** @brief Small interface class for loading trained serialized models of different dnn-frameworks. */
+    /**
+     * @deprecated Deprecated as external interface. Will be for internal needs only.
+     * @brief Small interface class for loading trained serialized models of different dnn-frameworks. */
     class CV_EXPORTS_W Importer : public Algorithm
     {
     public:
 
         /** @brief Adds loaded layers into the @p net and sets connections between them. */
-        CV_WRAP virtual void populateNet(Net net) = 0;
+        CV_DEPRECATED CV_WRAP virtual void populateNet(Net net) = 0;
 
-        virtual ~Importer();
+        CV_DEPRECATED virtual ~Importer();
```

Comment:
I think, destructors, which are called implicitly most of the time, should not be declared as deprecated

Reply:
Thanks! Fixed.